src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule.java

```diff
@@ -18,30 +32,107 @@
+    }
+
+    public void init(String profile, IndexReader indexReader, String subset, UserLog userLog, GoGoDuckConfig goGoDuckConfig) {
         this.profile = profile;
         this.indexReader = indexReader;
         this.subset = new GoGoDuckSubsetParameters(subset);
         this.userLog = userLog;
+        this.goGoDuckConfig = goGoDuckConfig;
     }
 
     public URIList getUriList() throws GoGoDuckException {
         return indexReader.getUriList(profile, TIME_FIELD, URL_FIELD, subset);
     }
 
+    public boolean postProcessingRequired() {
+        return postProcessingRequired.booleanValue();
+    }
+
+    public boolean isTimeUnlimited() {
+        return timeUnlimited.booleanValue();
+    }
+
+    public void setup(File file) {
+        boolean postProcessingRequired = false;
+        boolean timeUnlimited = true;
+        try {
+            String location = file.getAbsolutePath();
+            Set<NetcdfDataset.Enhance> enhanceMode = new HashSet<NetcdfDataset.Enhance>();
+            enhanceMode.add(NetcdfDataset.Enhance.ScaleMissing);
+            GridDataset gridDs = GridDataset.open(location, enhanceMode);
+            for (Variable var : gridDs.getNetcdfDataset().getVariables()) {
+                VariableDS vds = new VariableDS(var.getGroup(), var, true);
+                if (!postProcessingRequired && vds.hasScaleOffset()) {
+                    postProcessingRequired = true;
+                }
+
+                if (timeUnlimited && vds.isUnlimited()) {
+                    timeUnlimited = false;
+                }
+
+                if (postProcessingRequired && !timeUnlimited) {
+                    break;
+                }
+            }
+            this.postProcessingRequired = postProcessingRequired;
+            this.timeUnlimited = timeUnlimited;
+            gridDs.close();
+        } catch (IOException e) {
+            logger.error(e.getMessage());
+            throw new GoGoDuckException(e.getMessage());
+        }
+    }
+
     public void postProcess(File file) {
-        return;
+        try {
+            if (!postProcessingRequired()) {
+                return;
+            }
+
+            File tmpFile = File.createTempFile("ncpdq", ".nc");
+
+            List<String> command = new ArrayList<String>();
+            command.add(goGoDuckConfig.getNcpdqPath());
+            command.add("-O");
+            command.add("-U");
+            command.add(file.getAbsolutePath());
+            command.add(tmpFile.getAbsolutePath());
+
+            logger.info(String.format("Unpacking file (ncpdq) '%s' to '%s'", file.toPath(), tmpFile.toPath()));
+            GoGoDuck.execute(command);
+
+            Files.delete(file.toPath());
+            Files.move(tmpFile.toPath(), file.toPath());
+        } catch (Exception e) {
+            throw new GoGoDuckException(String.format("Could not run ncpdq on file '%s'", file.toPath()));
+        }
     }
 
     public List<String> ncksExtraParameters() {
```

Comment:
This was only used when time dimensions are not unlimited which we are handling elsewhere now - this option can be removed now?